src/domain/map/components/HankeSidebar/HankeSidebar.tsx

```diff
@@ -1,42 +1,106 @@
+import { formatToFinnishDate } from '../../../../common/utils/date';
+import { TFunction } from 'i18next';
+import HaittaIndexes from '../../../common/haittaIndexes/HaittaIndexes';
+import api from '../../../api/api';
+import { useQuery } from 'react-query';
+import axios from 'axios';
+import useHankeViewPath from '../../../hanke/hooks/useHankeViewPath';
+import { ROUTES } from '../../../../common/types/route';
+import { Link } from 'hds-react';
+
+function omistajaNimet(
+  omistajat: HankeYhteystieto[],
+  t: TFunction<'translation', undefined>,
+): string {
+  const names = omistajat.map((yhteystieto) =>
+    yhteystieto.nimi == undefined || yhteystieto.nimi.length > 0
+      ? yhteystieto.nimi
+      : t('hankeSidebar:labels:yksityishenkilo'),
+  );
+  const uniqueNames = Array.from(new Set(names));
+  return uniqueNames.join(', ');
+}
+
+const getHankkeet = async () => {
+  const { data } = await api.get<HankeData[]>(`/hankkeet`, {
+    params: {
+      geometry: false,
+    },
+  });
+  return data;
+};
+
+function useHankkeet() {
+  return useQuery<HankeData[] | null>(
+    ['hankkeet'],
+    async () => {
+      try {
+        return await getHankkeet();
+      } catch (error: unknown) {
+        if (
+          (axios.isAxiosError(error) && error.response?.status === 401) ||
+          (error instanceof Error && error.message === 'No token')
+        ) {
+          return null;
+        }
+        throw error;
+      }
+    },
+    {
+      enabled: true,
+    },
+  );
+}
 
 type SectionProps = {
   title: string;
-  content: string;
+  content: string | ReactNode;
 };
 
 const SidebarSection: React.FC<React.PropsWithChildren<SectionProps>> = ({ title, content }) =>
   title && title !== '' && content && content !== '' ? (
-    <>
+    <Box paddingBottom="var(--spacing-2-xs)">
       <Text tag="h3" styleAs="h6" weight="bold" spacingBottom="2-xs">
         {title}
       </Text>
-      <Text tag="p" styleAs="body-s" spacingBottom="2-xs">
-        {content}
-      </Text>
-    </>
+      {typeof content === 'string' ? (
+        <Text tag="p" styleAs="body-s" spacingBottom="2-xs">
+          {content}
+        </Text>
+      ) : (
+        <>{content}</>
+      )}
+    </Box>
   ) : null;
 
 type Props = {
   hanke: HankeData;
+  hankealueNimi: string;
   isOpen: boolean;
   handleClose: () => void;
 };
 
-const HankeSidebar: React.FC<React.PropsWithChildren<Props>> = ({ hanke, isOpen, handleClose }) => {
+const HankeSidebar: React.FC<React.PropsWithChildren<Props>> = ({
+  hanke,
+  hankealueNimi,
+  isOpen,
+  handleClose,
+}) => {
   const { t } = useTranslation();
-
+  const { data: hankkeet } = useHankkeet();
+  const isUsersHanke = hankkeet?.some((h) => h.hankeTunnus === hanke.hankeTunnus);
   const tyomaaTyyppiContent = hanke.tyomaaTyyppi.length
     ? hanke.tyomaaTyyppi.map((tyyppi) => t(`hanke:tyomaaTyyppi:${tyyppi}`)).join(', ')
     : '-';
+  const hankealue = hanke.alueet.find((ha) => ha.nimi === hankealueNimi)!;
```

Comment:
This won't work if there are multiple hanke alueet with same name which is possible.

Reply:
I changed it to use ids instead of names